### PR TITLE
feat(docs): Add Cardinality Processor doc

### PR DESCRIPTION
**Pull request by crazyuploader**
## Pull Request

---

### 📄 Summary
This PR adds doc for Cardinality Processor.

#### Issues closed by this PR
Closes https://github.com/SigNoz/growth-pod/issues/819

Doc: [Link](https://staging.signoz.io/docs/metrics-management/cardinality-processor)

---

### ✅ Change Type
_Select all that apply_

- [ ] ✨ Feature
- [ ] 🐛 Bug fix
- [ ] ♻️ Refactor
- [ ] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

---

- Tests added/updated:
- Manual verification:
- Edge cases covered:

---

- Blast radius:
- Potential regressions:
- Rollback plan:

---

| Field | Value |
|------|-------|
| Deployment Type | Cloud / OSS / Enterprise |
| Change Type | Feature / Bug Fix / Maintenance |
| Description | User-facing summary |

---

### 📋 Checklist
- [ ] Tests added or explicitly not required
- [ ] Manually tested
- [ ] Breaking changes documented
- [ ] Backward compatibility considered

---

## 👀 Notes for Reviewers

---

### CODE REVIEW

File: data/docs-side-nav/main.json [hunk 1]
```diff
@@ -2083,6 +2083,11 @@
         "isExpanded": false,
         "route": "/docs/metrics-management/user-guides",
         "items": [
+          {
+            "type": "doc",
```
Comment (revmag) [inline data/docs-side-nav/main.json]: This should be in reference in opentelemetry collector 
https://signoz.io/docs/opentelemetry-collection-agents/opentelemetry-collector/tls-configuration/

File: data/docs/metrics-management/cardinality-processor.mdx [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+- **Metrics only.** The processor operates on metric data. Logs and traces are unaffected.
+- **Naming quirk.** The component built into the collector is named `cardinalityprocessor`, but the key you use in the `processors:` block of your config is `cardinality_guardian`.
+- **Where it runs.** You add it to a collector that *you* run. SigNoz Cloud's managed collector cannot load custom processors, so this sits in your own collector, upstream of the OTLP export to SigNoz.
+
+## How it works
+
+Instead of a fixed limit on total series, the processor watches the **rate of growth** of unique values for each `(metric, attribute)` pair:
+
+- The processor divides time into fixed windows called **epochs** (`epoch_duration_seconds`, default 300s).
+- For each pair, it estimates how many *new* unique values appear within an epoch using <a href="https://en.wikipedia.org/wiki/HyperLogLog" target="_blank" rel="noopener noreferrer nofollow">HyperLogLog++</a> sketches (roughly 2 KB of memory per tracked pair).
```
Comment (revmag) [inline data/docs/metrics-management/cardinality-processor.mdx]: Here many unncessary info is there, what you can do, is keep it very simple initially, and then add a detailed info section on how it works

**Review by revmag — APPROVED** [review]
